TypeError: unsupported operand type(s) for |: 'NoneType' and 'NoneType'

### Long story short

Trying out the new version 1.44.3 results in a type error.
See our CI run here:
https://github.com/gefyrahq/gefyra/actions/runs/23451263771/job/68229005321?pr=908


### Kopf version

1.44.3

### Kubernetes version

_No response_

### Python version

3.11

### Code

```python

```

### Logs

```none
venv/lib/python3.11/site-packages/kopf/__init__.py:10: in <module>
    from kopf import (
.venv/lib/python3.11/site-packages/kopf/on.py:19: in <module>
    from kopf._cogs.structs import dicts, references, reviews
.venv/lib/python3.11/site-packages/kopf/_cogs/structs/dicts.py:9: in <module>
    from kopf._cogs.helpers import thirdparty
.venv/lib/python3.11/site-packages/kopf/_cogs/helpers/thirdparty.py:84: in <module>
    class KubernetesModelAsync(abc.ABC):
.venv/lib/python3.11/site-packages/kopf/_cogs/helpers/thirdparty.py:93: in KubernetesModelAsync
    def metadata(self) -> V1ObjectMetaAsync | None:
                          ^^^^^^^^^^^^^^^^^^^^^^^^
E   TypeError: unsupported operand type(s) for |: 'NoneType' and 'NoneType'
```

### Additional information

_No response_

--- a/kopf/_cogs/helpers/thirdparty.py
+++ b/kopf/_cogs/helpers/thirdparty.py
@@ -4,6 +4,8 @@ Type definitions from optional 3rd-party libraries, e.g. pykube-ng & kubernetes.
 This utility does all the trickery needed to import the libraries if possible,
 or to skip them and make typing/runtime dummies for the rest of the codebase.
 """
+from __future__ import annotations
+
 import abc
 from typing import Any, Protocol
 

